relayer/processor/types.go

```diff
@@ -77,6 +77,15 @@ type IBCMessagesCache struct {
 	ChannelHandshake    ChannelMessagesCache
 }
 
+// Clone makes a deep copy of an IBCMessagesCache.
+func (c IBCMessagesCache) Clone() IBCMessagesCache {
+	x := NewIBCMessagesCache()
```

Comment:
This is going to initialize all the maps to a default length. It seems like it would be a worthwhile tradeoff to correctly presize the maps, i.e.

```go
x := IBCMessagesCache{
  PacketFlow: make(ChannelPacketMessagesCache, len(c.PacketFlow)),
  // etc.
}
// Then merge individual fields.
```

This could avoid some unnecessary garbage creation.

Reply:
good catch, done